src/generic_ui/scripts/ui.ts

```diff
@@ -134,6 +134,25 @@ export interface NotificationData {
   unique ?:string;
 }
 
+var PROXYING_SESSION_ID_LENGTH = 16;
+
```

Comment:
Ah, I just realised one-time connections take a slightly different path; yes, it would be better to generate the ID in remote connection. I've just done this. PTAL, thanks!